src/test/java/org/example/jvspringbootfirstbook/repository/cart/ShoppingCartRepositoryTest.java

```diff
@@ -0,0 +1,101 @@
+package org.example.jvspringbootfirstbook.repository.cart;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.example.jvspringbootfirstbook.model.Book;
+import org.example.jvspringbootfirstbook.model.CartItem;
+import org.example.jvspringbootfirstbook.model.ShoppingCart;
+import org.example.jvspringbootfirstbook.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class ShoppingCartRepositoryTest {
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @Test
+    @DisplayName("""
+            Should return the ShoppingCart associated with the User having ID 2
+            """)
+    @Sql(scripts = {"classpath:db/cart/repository/delete_existing_carts.sql",
+            "classpath:db/cart/repository/add_shopping_cart_with_necessities.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {"classpath:db/cart/repository/delete_existing_carts.sql"},
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void findShoppingCartByUser_UserWithIdEqualOne_returnsShoppingCart() {
+        Long userId = 1L;
+        ShoppingCart testShoppingCart = getTestShoppingCart();
+        ShoppingCart actual = shoppingCartRepository
+                .findShoppingCartByUser(userId);
+        Assertions.assertAll(
+                () -> Assertions.assertEquals(testShoppingCart.getId(),
+                        actual.getId()),
+                () -> Assertions.assertEquals(testShoppingCart.getUser().getEmail(),
+                        actual.getUser().getEmail()),
+                () -> Assertions.assertEquals(
+                        takeItemsAndGetString(testShoppingCart),
+                        takeItemsAndGetString(actual))
+        );
```

Comment:
Consider using `assertIterableEquals`, here you can read more about different Assertions in JUnit:
https://www.baeldung.com/junit-assertions#:~:text=The%20assertIterableEquals%20asserts%20that%20the,in%20order%20to%20be%20equal.